exchangelib/account.py

```diff
@@ -747,6 +756,69 @@ def delegates(self):
         """Return a list of DelegateUser objects representing the delegates that are set on this account."""
         return list(GetDelegate(account=self).call(user_ids=None, include_permissions=True))
 
+    @property
+    def rules(self):
+        """Return a list of Rule objects representing the rules that are set on this account."""
+        try:
+            elems = GetInboxRules(account=self).call()
+            return list(elems)
+        except MalformedResponseError as exc:
+            empty_info = f"No {{{MNS}}}InboxRules elements in ResponseMessage"
+            if empty_info in str(exc):
+                return []
+            raise exc
+
+    def get_rules(self, generator=False) -> List[Rule]:
+        """Retrieve Inbox rules in the identified user's mailbox.
+
+        :param generator: If True, return a generator instead of a list. Default is False.
+        :return: A list of Rule objects if generator is False, else a generator.
+        """
+        if generator:
+            # If the generator parameter is True, return a generator.
+            return GetInboxRules(account=self).call()
+        return self.rules
+
+    def create_rule(self, rule: Rule):
+        """Create an Inbox rule in a user's mailbox in the Exchange store.
+
+        :param rule: The rule to create with display_name as the key at least.
+        :return: The rule ID of the created rule. If failed, raise an error.
+        """
+        create_rule_service = CreateInboxRule(account=self)
+        create_rule_service.get(rule=rule, remove_outlook_rule_blob=True)
+        # After creating the rule, query all rules,
+        # find the rule that was just created, and return its ID.
+        rules = self.get_rules(generator=True)
+        for i in rules:
+            if i.display_name == rule.display_name:
+                return i.rule_id
+        raise ResponseMessageError("Failed to create rule!")
+
+    def set_rule(self, rule: Rule):
+        """Modify an Inbox rule in a user's mailbox in the Exchange store.
+
+        :param rule: The rule to set with rule_id as the key at least.
+        :return: None if success, else raise an error.
+        """
+        return SetInboxRule(account=self).get(rule=rule)
+
+    def delete_rule(self, rule_name: Optional[str] = None, rule_id: Optional[str] = None):
```

Comment:
I think it's better if this method just takes a `Rule` object instead. We can let the user get a hold of the rule ID via `Account.get_rules()` if they only have the name.